scripts/o3de/o3de/android.py

```diff
@@ -0,0 +1,608 @@
+                                                      strip_debug_symbols=strip_debug,
+                                                      src_pak_file_path=src_bundle_pak_subfolder,
+                                                      oculus_project=is_oculus_project)
+        apg.execute()
+
+    except android_support.AndroidToolError as err:
+        logger.error(str(err))
+        return 1
+    else:
+        return 0
+
+
+def add_args(subparsers) -> None:
+    """
+    add_args is called to add subparsers for the following commands to the central o3de.py command processor:
+
+    - android_configure
+    - android_generate
+
+    :param subparsers: the caller instantiates subparsers and passes it in here
+    """
+
+    # Read from the android config if possible to try to display default values
+    try:
+        project_name, project_path = command_utils.resolve_project_name_and_path()
+        android_config = android_support.get_android_config(project_name=project_name)
+    except (android_support.AndroidToolError, command_utils.O3DEConfigError):
+        logger.debug(f"No project detected at {os.getcwd()}, default settings from global config.")
+        project_name = None
+        android_config = android_support.get_android_config(project_name=None)
+
+    #
+    # Configure the subparser for 'android-configure'
+    #
+
+    # Generate the epilog string to describe the settings that can be configured
+    epilog_lines = ['Configure the default values that control the android helper scripts for O3DE.',
+                    'The list settings that can be set are:',
+                    '']
+    max_key_len = 0
+    for setting in android_config.setting_descriptions:
+        max_key_len = max(len(setting.key), max_key_len)
+    max_key_len += 4
+    for setting in android_config.setting_descriptions:
+        setting_line = f"{'* ' if setting.is_password else '  '}{setting.key: <{max_key_len}} {setting.description}"
+        epilog_lines.append(setting_line)
+    epilog_lines.extend([
+        '',
+        '* Denotes password settings that can only be set using the --set-password command'
+    ])
+
+    epilog = '\n'.join(epilog_lines)
+
+    android_configure_subparser = subparsers.add_parser(O3DE_COMMAND_CONFIGURE,
+                                                        help='Configure the Android platform settings for generating, building, and deploying Android projects.',
+                                                        epilog=epilog,
+                                                        formatter_class=argparse.RawTextHelpFormatter)
+
+    android_configure_subparser.add_argument('--global', default=False, action='store_true',
+                                             help='Used with the configure command, specify whether the settings are project local or global.')
+    android_configure_subparser.add_argument('-p', '--project', type=str, required=False,
```

Comment:
perhaps in a future PR it'd be better to go with a project-path setting because with versioning now supported, it's possible the user could have a project registered multiple times but different versions of it

Reply:
or maybe support both? Like we can check if `project` is an absolute path (or even a subpath), otherwise assume its just a name and try to resolve the path (with an info or warning)

Reply:
Updated to support project path (first priority) and then project name. Will update the document as well